### ISSUE TITLE
resolveConfig is slow when deals with thousands of theme

### ISSUE THREAD

**Issue by hardfist** [tailwindlabs/tailwindcss#5651]
**What version of Tailwind CSS are you using?**

2.2.16

**What build tool (or framework if it abstracts the build tool) are you using?**

postcss

**What version of Node.js are you using?**

For example: v14.18.0

**What browser are you using?**

N/A

**What operating system are you using?**

 macOS

**Reproduction URL**

**Describe your issue**

when deals with thousands of theme, the following implementation is very in efficient
```js
const configUtils = {
  negative(scale) {
    return Object.keys(scale).filter(key => scale[key] !== '0').reduce((negativeScale, key) => ({ ...negativeScale,
      [`-${key}`]: (0, _negateValue.default)(scale[key]) // ...negativeScale is very slow when negativeScale has lots of keys
    }), {});
  },

  breakpoints(screens) {
    return Object.keys(screens).filter(key => typeof screens[key] === 'string').reduce((breakpoints, key) => ({ ...breakpoints,
      [`screen-${key}`]: screens[key]
    }), {});
  }
};
```

**Comment by hardfist** [comment]
# use spread
![image](https://user-images.githubusercontent.com/8898718/135406053-2593e462-95f1-4c69-b606-0e0fcf5cc274.png)

# not use spread
![image](https://user-images.githubusercontent.com/8898718/135405905-a6323a78-ee92-4010-8b94-e849dab24fe7.png)


v8 has an issue related object spread has bad performance  https://bugs.chromium.org/p/v8/issues/detail?id=11536&q=spread%20slow&can=2

**Comment by adamwathan** [comment]
Fixed in #5652
